src/Backups/BackupImpl.cpp

```diff
@@ -971,6 +989,18 @@ void BackupImpl::increaseUncompressedSize(const FileInfo & info)
         increaseUncompressedSize(info.size - info.base_size);
 }
 
+void BackupImpl::increaseProcessedSize(UInt64 file_size) const
+{
+    std::lock_guard lock{mutex};
```

Comment:
`increaseProcessedSize()` is called from `writeBackupMetadata` while the mutex is already locked. That means a deadlock, that's why the tests failed.

Reply:
Thank you very much!